### ISSUE TITLE
Jazz Jackrabbit 2 package in the Debian (+ Ubuntu, etc.) official repositories

### ISSUE THREAD

**Issue by vv221** [deathkiller/jazz2-native#87]
Disclaimer: This is not a bug report, sorry if this is not the best channel to get in touch. I am not a Discord user so I could not join you through this channel.

---

I plan to start working soon on integrating Jazz² Resurrection into the official Debian repositories, so all Debian users could install it by running `apt install jazz2-resurrection` without having to go through deat.tk nor GitHub. This would in turn make it available for Debian derivatives like Ubuntu or Linux Mint.

If you wish to follow all activity on this front, I will keep the following Debian bug report updated: https://bugs.debian.org/1088176

If some maintainers of Jazz² Resurrection would like to be kept in the loop for anything Jazz² Resurrection-related on the Debian side, please share an e-mail with me and I’ll make sure to include you in all further discussion.

Obviously, feel free to ask for any kind of detail, and I’ll answer to the best of my abilities.

**Comment by vv221** [comment]
It’s pleasing to see such a quick and warm reaction, I’ll make sure to send you a ping if I get any problem.

I expect to start working on this package in a couple days, maybe a couple weeks, depending on my free time. When I get to it I’ll share my progress here to keep you in the loop.

**Comment by vv221** [comment]
Just a quick ping to say I have not forgotten my plan to package this engine for Debian.

Right now I’m waiting for the previous engine I packaged (https://github.com/bvschaik/julius/issues/739) to enter the Debian repositories, then Jazz² Resurrection is the next in line.

Sorry for the « (…) in a couple days » in my previous message, I shifted my priorities when I noticed Julius, the other engine, had most of the preliminary work already done by other Debian contributors.

**Comment by veddevv** [comment]
Maybe it will be a copyright problem if you distribute it unofficially, idk

**Comment by deathkiller** [comment]
The original game data is not included. It must be supplied externally by the user, so there is no copyright problem.

Also, DEB packages are already generated by the build system, so hopefully it won't be difficult to add them to the official repositories (maybe with some adjustments) either.

**Comment by vv221** [comment]
Checking the license of all source files is part of the process when including a software into Debian, so if there is any problem on this front (I doubt we should find any) we would probably spot it and report about it here so it can be fixed.

Of course this is all about distributing only the engine, users would still have to provide the data by some supported mean. My motivation for packaging it is related to that, as I already provide a tool to extract the game data and plug it into the game engine — https://wiki.debian.org/Games/PlayIt — and this tool would benefit greatly from the ability to pick up Jazz² Resurrection directly from the distribution repositories, instead of asking the users to download it on their own.

**Comment by vv221** [comment]
I see you provide two variants of the engine:
- one relying on libglfw.so.3
- one relying on libSDL2-2.0.so.0

Do you have a preference on which one should be packaged into Debian?

In my experience they both work perfectly, but I see no real reason to include both and confuse the user with a choice of rendering engine when they want to install Jazz² Resurrection.

**Comment by deathkiller** [comment]
Yes, I would include only the SDL2 variant, it has better gamepad support (and other than that it's almost the same), and it's already preferred also on other platforms (e.g., Flathub)

**Comment by vv221** [comment]
Good, SDL2 it is then!

Another question: I see you included a copy of jsoncpp source in your repository. Did you customize it in some way, or could a system-provided jsoncpp be used during the build instead?

**Comment by deathkiller** [comment]
It's my own fork of jsoncpp, it's definitely not compatible/interchangeable with the system-provided one. Basically every `.h`/`.cpp` file in the repo is customized.

**Comment by vv221** [comment]
Duly noted. It comes with the same GPL v3.0 license than the rest of the source?

And just to be sure: all third-party sources that can be replaced with system-provided development libraries are in the dedicated repository jazz2-libraries? No third-party code is to be found in the jazz2-native repository?

**Comment by deathkiller** [comment]
Yes, everything in the `jazz2-native` repo is licensed as GPL v3.0 as stated in https://github.com/deathkiller/jazz2-native/blob/master/LICENSE.

I'm not sure what do you mean. `jazz2-libraries` repository doesn't contain sources, it contains only pre-compiled libraries in case there are no better libraries present in the system. `jazz2-libraries` has the lowest search priority (if you don't disable it completely). So I think you should use system-provided libraries instead, if you can.

Also, all dependencies are listed here (https://deat.tk/jazz2/docs/third-party.html). Basically all single-header libraries (+ jsoncpp) are included directly, everything else is linked. Not all of them are used on Linux in the default build configuration though. Half of them are reserved for future use or disabled completely.

**Comment by vv221** [comment]
> I'm not sure what do you mean. `jazz2-libraries` repository doesn't contain sources, it contains only pre-compiled libraries (…)

My mistake, I was indeed talking about the pre-built libraries.

I think you answered all interrogations I had, I should now know everything I need to start on the actual packaging.

**Comment by vv221** [comment]
Done, Jazz² Resurrection is now provided by Debian: https://tracker.debian.org/pkg/jazz2-native

Version 3.3.0 for now, because it is the one I packaged at the time and its processing ended up delayed for several months for external reasons. I’m going to update it to 3.5.0 soon, maybe even tonight, this time (and for further update) no delay is going to happen.

Welcome into Debian repositories!

**Comment by deathkiller** [comment]
That's nice. Thank you!

**Comment by vv221** [comment]
And now it looks like it reached Ubuntu: https://launchpad.net/ubuntu/+source/jazz2-native
Ubuntu "Resolute Racoon" (26.04) should be the first Ubuntu LTS release to ship Jazz² Resurrection.
